main/solution/prepare-devops-acc/config/infra/cloudformation.yml

```diff
@@ -0,0 +1,33 @@
+Resources:
+  DevOpsRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      RoleName: ${self:custom.settings.devopsRoleName}
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: 'Allow'
+            Principal:
+              AWS:
+                - !Sub ${self:custom.settings.workflowLoopRunnerRoleArn}
+                - !Sub ${self:custom.settings.apiHandlerRoleArn}
+            Action:
+              - 'sts:AssumeRole'
+            Condition:
+              StringEquals:
+                sts:ExternalId: ${self:custom.settings.externalId}
+      Policies:
+        - PolicyName: AllowAMISharingtoSWBMainAccount
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Sid: EC2AMISharingAccess
+                Effect: Allow
+                Action:
+                  - ec2:ModifyImageAttribute
+                  - ec2:DescribeImages
+                  - appstream:UpdateImagePermissions
+                Resource: "*"
```

Comment:
Could we restrict this further, maybe by using a tag?

Reply:
Since the AMIs created using SWB Packer does not have any tags and also the user can use any custom AMI IDs (Like the ones created by RG repo also). We have not restricted this further.